framework/web/XmlResponseFormatter.php

```diff
@@ -95,18 +115,22 @@ protected function buildXml($element, $data)
                 if (is_int($name) && is_object($value)) {
                     $this->buildXml($element, $value);
                 } elseif (is_array($value) || is_object($value)) {
-                    $child = new DOMElement($this->getValidXmlElementName($name));
+                    $child = $this->dom->createElement($this->getValidXmlElementName($name));
                     $element->appendChild($child);
                     $this->buildXml($child, $value);
                 } else {
-                    $child = new DOMElement($this->getValidXmlElementName($name));
+                    $child = $this->dom->createElement($this->getValidXmlElementName($name));
+                    $child->appendChild($this->dom->createTextNode($this->formatScalarValue($value)));
                     $element->appendChild($child);
-                    $child->appendChild(new DOMText($this->formatScalarValue($value)));
                 }
             }
         } elseif (is_object($data)) {
             if ($this->useObjectTags) {
-                $child = new DOMElement(StringHelper::basename(get_class($data)));
+                $name = StringHelper::basename(get_class($data));
+                if ($this->objectTagToLowercase) {
+                    $name = strtolower($name);
```

Comment:
Won't work well with UTF.

Reply:
@samdark as Yii at all, for example https://github.com/yiisoft/yii2/blob/bcd5b48492219efc58e1d11c9ddb61c0df0dc370/framework/console/controllers/MigrateController.php#L409